File: sub-web-model/src/main/java/org/github/base/entity/SysUserEntity.java
package org.github.base.entity;

import org.github.base.Model;
import java.time.LocalDateTime;
import com.baomidou.mybatisplus.annotation.*;
import java.io.Serializable;
import javax.annotation.Nullable;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * <p>
 * 系统用户
 * </p>
 *
 * @author JYD_XL
 * @since 2021-05-07
 */
@SuppressWarnings("all")
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_user")
@ApiModel(value="SysUserEntity对象", description="系统用户")
public class SysUserEntity extends Model<SysUserEntity> {

  /** UID */
  private static final long serialVersionUID = 1L;

  @TableId(value = "user_id", type = IdType.AUTO)
  private @Nullable Long userId;

  /** 用户名 */
  @ApiModelProperty(value = "用户名")
  @TableField("username")
  private @Nullable String username;

  /** 密码 */
  @ApiModelProperty(value = "密码")
  @TableField("password")
  private @Nullable String password;

  /** 邮箱 */
  @ApiModelProperty(value = "邮箱")
  @TableField("email")
  private @Nullable String email;

  /** 手机号 */
  @ApiModelProperty(value = "手机号")
  @TableField("mobile")
  private @Nullable String mobile;

  /** 状态  0：禁用   1：正常 */
  @ApiModelProperty(value = "状态  0：禁用   1：正常")
  @TableField("status")
  private @Nullable Integer status;

  /** 创建者ID */
  @ApiModelProperty(value = "创建者ID")
  @TableField("create_user_id")
  private @Nullable Long createUserId;

  /** 创建时间 */
  @ApiModelProperty(value = "创建时间")
  @TableField("create_time")
  private @Nullable LocalDateTime createTime;

  @Override
  protected Serializable pkVal() {
    return this.userId;
  }

}

File: sub-mysql-mydb-entity/src/main/java/org/github/mysql/mydb/base/entity/SysUserEntity.kt
@file:Suppress("UnusedImport", "unused")

package org.github.mysql.mydb.base.entity

import com.baomidou.mybatisplus.annotation.FieldFill
import com.baomidou.mybatisplus.annotation.IdType
import com.baomidou.mybatisplus.annotation.TableField
import com.baomidou.mybatisplus.annotation.TableId
import com.baomidou.mybatisplus.annotation.TableName
import java.io.Serializable
import java.time.LocalDateTime
import org.github.base.Entity

/**
 * 用户表
 *
 * @author JYD_XL
 */
@TableName("sys_user")
open class SysUserEntity : Entity() {

    /** 主键 */
    @TableId(value = "user_id", type = IdType.ASSIGN_ID)
    open var userId: Long? = null

    /** 用户名 */
    @TableField("username")
    open var username: String? = null

    /** 密码 */
    @TableField("password")
    open var password: String? = null

    /** 盐 */
    @TableField("salt")
    open var salt: String? = null

    /** 启用状态 0：禁用 1：启用 */
    @TableField("enabled")
    open var enabled: Byte? = null

    /** 昵称 */
    @TableField("nickname")
    open var nickname: String? = null

    /** 邮箱 */
    @TableField("email")
    open var email: String? = null

    /** 手机号 */
    @TableField("mobile")
    open var mobile: String? = null

    /** 创建人id */
    @TableField(value = "creator_id", fill = FieldFill.INSERT)
    open var creatorId: Long? = null

    /** 创建时间 */
    @TableField("create_time")
    open var createTime: LocalDateTime? = null

    /** 修改人id */
    @TableField("modifier_id")
    open var modifierId: Long? = null

    /** 修改时间 */
    @TableField("modify_time")
    open var modifyTime: LocalDateTime? = null

    /** 删除状态 0：未删除 1：已删除 */
    @TableField("deleted")
    open var deleted: Byte? = null

    companion object {

        private const val serialVersionUID = 1L

        const val USER_ID : String = "user_id"

        const val USERNAME : String = "username"

        const val PASSWORD : String = "password"

        const val SALT : String = "salt"

        const val ENABLED : String = "enabled"

        const val NICKNAME : String = "nickname"

        const val EMAIL : String = "email"

        const val MOBILE : String = "mobile"

        const val CREATOR_ID : String = "creator_id"

        const val CREATE_TIME : String = "create_time"

        const val MODIFIER_ID : String = "modifier_id"

        const val MODIFY_TIME : String = "modify_time"

        const val DELETED : String = "deleted"

    }

    override fun pkVal(): Serializable? {
        return userId
    }

    override fun toString(): String {
        return "SysUserEntity{" +
        "userId=" + userId +
        ", username=" + username +
        ", password=" + password +
        ", salt=" + salt +
        ", enabled=" + enabled +
        ", nickname=" + nickname +
        ", email=" + email +
        ", mobile=" + mobile +
        ", creatorId=" + creatorId +
        ", createTime=" + createTime +
        ", modifierId=" + modifierId +
        ", modifyTime=" + modifyTime +
        ", deleted=" + deleted +
        "}"
    }

}
